rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1851,6 +1857,17 @@ private BlockToMineBuilder getBlockToMineBuilder() {
         return blockToMineBuilder;
     }
 
+    private MinGasPriceProvider getMinGasPriceProvider() {
+        if (minGasPriceProvider == null) {
+            long minGasPrice = getRskSystemProperties().minerMinGasPrice();
+            DefaultMinGasPriceProvider defaultMinGasPriceProvider = new DefaultMinGasPriceProvider(minGasPrice);
```

Comment:
As I said, I am not sure if we need create this object here, I thought we had access to the config during the provider creation using the factory. Reflect about this, and let me know you opinion. 👍